packages/app/pages/api/ethpass.ts

```diff
@@ -0,0 +1,54 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import type { ContractSendMethod } from "web3-eth-contract";
+import axios from "axios";
+import { getWeb3 } from "../../components/utils";
+import { getAbiFromJson } from "../../components/constants";
+import passportJson from "@cabindao/nft-passport-contracts/artifacts/contracts/Passport.sol/Passport.json";
+
+export default function handler(req: NextApiRequest, res: NextApiResponse) {
+  const { address, network, tokenId, signature, signatureMessage } =
+    typeof req.body === "string" ? JSON.parse(req.body) : req.body;
+  switch (req.method) {
+    case "POST":
+      const web3 = getWeb3(network);
+      const contract = new web3.eth.Contract(
+        getAbiFromJson(passportJson),
+        address
+      );
+      return (contract.methods.symbol() as ContractSendMethod)
+        .call()
+        .then((description) =>
+          axios.post(
+            "https://api.ethpass.xyz.com/api/v1/pass",
+            {
+              contractAddress: address,
+              tokenId,
+              signature,
+              signatureMessage,
+              pass: {
+                type: "generic",
+                description,
+              },
+            },
+            {
+              headers: {
+                "X-API-KEY": process.env.ETHPASS_API_KEY || "",
+                "x-api-scope": "pass.com.ethpass",
+              },
+            }
+          )
+        )
+        .then((r) => res.status(200).json(r.data))
```

Comment:
This shape of the response payload currently is `{ id, buffer}` where `id` is the unique id of the pass created, and `buffer` is the pass file in a raw format so you can choose how you want to distribute it. How are you envisioning users installing the pass?

Reply:
we will probably upload it to IPFS. Could we also display it as a QR code that can be scanned right in the checkout?

Reply:
Yes, you can create a QR code with the file url. Here's how it's done on our demo.
```
import QRCode from 'qrcode'

const canvas = document.getElementById('canvas');
const fileURL = "https://"

QRCode.toCanvas(canvas, fileURL, function (error) {
  if (error) console.error(error);
});
```

There are a few caveats when dealing with IPFS. We can't specify content header types when trying to download files from public gateways so this has been causing some issues on older iOS devices. I'm considering adding IPFS as a storage option out the box to onboard users quicker alongside a proxy that will take care of setting the content header types properly. 

In the meantime, if you'd like to test uploading the buffer to IPFS you can use the following code.
```
export const uploadIpfs = async (buffer: Buffer) => {
  try {
    const passName = `DemoPass-${Math.random()}.pkpass`;
    const auth = 'Basic ' + Buffer.from(projectId + ':' + projectSecret).toString('base64');
    const client = create({
      host: 'ipfs.infura.io',
      port: 5001,
      protocol: 'https',
      headers: {
        authorization: auth,
      },
    });
    const response = await client.add(buffer, { pin: false });
    return `https://ipfs.infura.io/ipfs/${response.path}?filename=${passName}`;
  } catch (err) {
    console.log('## IPFS Upload error:', err);
    throw err;
  }
};